src/idlcxx/src/backendCpp11Utils.c

```diff
@@ -200,8 +201,8 @@ get_cpp11_fully_scoped_name(const idl_node_t *node)
   scoped_enumerator[0] = '\0';
   while(nr_scopes)
   {
-    strncat(scoped_enumerator, "::", scoped_enumerator_len);
-    strncat(scoped_enumerator, scope_names[--nr_scopes], scoped_enumerator_len);
+    ddsrt_strlcat(scoped_enumerator, "::", scoped_enumerator_len);
+    ddsrt_strlcat(scoped_enumerator, scope_names[--nr_scopes], scoped_enumerator_len);
```

Comment:
Can't use ddsrt in idlcxx

Reply:
@k0ekk0ek it is perhaps wise to reconsider that position, especially in the light of https://github.com/eclipse-cyclonedds/cyclonedds-cxx/pull/19#discussion_r492791663 which at least suggests that this is a mistake so easily made that it would be better for it not to be considered a mistake.

The only complication I see is a (likely, future) circular dependency caused by linking idlc with libddsc, and having two definitions of the same symbols around if ddsrt is (statically) linked into idlc and idlc and libddsc are both shared libraries linked into the same process. Making ddsrt a shared library in its own right would solve both, but is a bit of barrier to LTO.

Another possible solution (one that I would be fine with) is linking idlc twice: once for bootstrapping libddsc wherein ddsrt is statically linked, and once against libddsc for distribution.

Reply:
I really prefer not to use `ddsrt` in `idlc` or `libidl` (this is about `idlcxx`, which is a separate library of course). But anyway, `ddsrt` must run on platforms that really don't offer much. The same is not true for platforms where the idl compiler will run. That doesn't mean we need to copy the code, (I/we)'ll think of something for that, but not having `ddsc` in `idl` ensures it's as small as possible. I'm hoping that at some point it'll pay off. For instance, we could have a webapp people can use to generate code. Which probably means Python (or similar) calling into C, at that point, I'm sure people prefer not to have `ddsc` loaded too(?) I also haven't let go of the idea of moving it into a separate project entirely. People may find a decent BSD-licensed IDL compiler very useful. For example, I would've really appreciated it if one would already have been available :sweat_smile:

Reply:
At the moment, I will remove the changes that require the `ddsrt` library.
This will mean that the file-wide warning suppressions for those files will have to come back as well, since single-line warning suppression pragmas cause warnings for linux compilers.

Reply:
Another use-case for a separate library (just popped into my mind) might be that we eventually use it to generate code for e.g. builtin topics. I'm not saying we will at some point, but it's not unthinkable either(?)